src/globus_sdk/services/auth/client/confidential_client.py

```diff
@@ -294,3 +294,107 @@ def oauth2_token_introspect(
         if include is not None:
             body["include"] = include
         return self.post("/v2/oauth2/token/introspect", data=body, encoding="form")
+
+    # TODO: It is a little strange to place this in this class but Auth treats
+    # client creation different based on how the client was authenticated.
```

Comment:
This doesn't strike me as a TODO. I recommend removing this comment here, and in the Native App code.